### PR TITLE
added Pyramid Stack , Ray Cast and modified something

### PR DESCRIPTION
**Pull request by cuit-zhaxin**
added  **Pyramid Stack** , **Ray Cast** and modified something

What I did?
- added the project to windows, so you can run in the win with vs
- modified **CCComponentPhysics2d.h** and **CCPhysicsManager.h** with `CC_DLL` added to them 
- modified functions `addGrossiniAtPosition`,  `makeBall`, `makeBox` and `makeTriangle`
- the effect of **Logo Smash** was improved because of the modification of the function `makeBall`
- added test case  **Pyramid Stack**
- added test case **Ray Cast** with something still needed to be fixed

### CODE REVIEW

File: plugin [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit ad539b56c5354c1083f479d4c8d275ae8c7fdfa7
```
Comment (minggo) [inline plugin]: Please don't modify submodule.

File: tests/cpp-tests/Classes/PhysicsTest/PhysicsComponentTest.cpp [hunk 2]
```diff
@@ -10,7 +10,9 @@ USING_NS_CC;
 PhysicsComponentTests::PhysicsComponentTests()
 {
     ADD_TEST_CASE(PhysicsComponentDemoLogoSmash);
-    ADD_TEST_CASE(PhysicsComponentDemoClickAdd);
+	ADD_TEST_CASE(PhysicsComponentDemoPyramidStack);
+	ADD_TEST_CASE(PhysicsComponentDemoClickAdd);
+	ADD_TEST_CASE(PhysicsComponentDemoRayCast);
```
Comment (minggo) [inline tests/cpp-tests/Classes/PhysicsTest/PhysicsComponentTest.cpp]: please replace tab with 4 spaces.

File: tests/cpp-tests/Resources/ccs-res [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 65f45cef163bea7f25d20fe9ba77f6af871656c0
+Subproject commit 7b3f21b5ad03445b907e0b713804fdc35f3c0959
```
Comment (minggo) [inline tests/cpp-tests/Resources/ccs-res]: Please don't modify submodule.

File: tests/cpp-tests/Classes/PhysicsTest/PhysicsComponentTest.cpp [hunk 4]
```diff
@@ -201,10 +208,12 @@ Sprite* PhysicsComponentDemo::makeTriangle(Vec2 point, Size size, int color, Phy
         triangle->setScaleY(size.height/43.5f);
     }
     
-    Vec2 vers[] = { Vec2(0, size.height/2), Vec2(size.width/2, -size.height/2), Vec2(-size.width/2, -size.height/2)};
-    
-    auto body = PhysicsBody::createPolygon(vers, 3, material);
-    addPhysicsComponent(triangle, body);
+    //Vec2 vers[] = { Vec2(0, size.height/2), Vec2(size.width/2, -size.height/2), Vec2(-size.width/2, -size.height/2)};
+	Vec2 vers[] = { Vec2(0, triangle->getContentSize().height / 2), Vec2(triangle->getContentSize().width / 2, -triangle->getContentSize().height / 2), Vec2(-triangle->getContentSize().width / 2, -triangle->getContentSize().height / 2) };
```
Comment (minggo) [inline tests/cpp-tests/Classes/PhysicsTest/PhysicsComponentTest.cpp]: Why modify like this?

Comment (cuit-zhaxin) [inline tests/cpp-tests/Classes/PhysicsTest/PhysicsComponentTest.cpp]: the values `size.heght` and `size.width` you used in `vers[]` can not correctly express the body's size

**Pull request merged**